File: fd-angular-ui/src/app/core/services/fd-account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { FdAccount, Transaction, Statement } from '../models/models';

@Injectable({
  providedIn: 'root'
})
export class FdAccountService {
  private apiUrl = 'http://localhost:8080/api/fd/account';
  private allAccountsUrl = 'http://localhost:8080/api/fd/accounts';

  constructor(private http: HttpClient) {}

  createAccount(request: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/create`, request);
  }

  getMyAccounts(): Observable<FdAccount[]> {
    return this.http.get<FdAccount[]>(`${this.allAccountsUrl}/my`);
  }

  getAllAccounts(): Observable<FdAccount[]> {
    return this.http.get<FdAccount[]>(`${this.allAccountsUrl}/all`);
  }

  getAccount(fdAccountNo: string): Observable<FdAccount> {
    return this.http.get<FdAccount>(`${this.apiUrl}/${fdAccountNo}`);
  }

  getTransactions(fdAccountNo: string): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}/${fdAccountNo}/transactions`);
  }

  getStatements(fdAccountNo: string): Observable<Statement[]> {
    return this.http.get<Statement[]>(`${this.apiUrl}/${fdAccountNo}/statements`);
  }

  withdraw(request: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/withdraw`, request);
  }

  manualClose(fdAccountNo: string): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/manual-close?fdAccountNo=${fdAccountNo}`, {});
  }
}

File: fd-microservice/src/main/java/com/bank/fd/service/FdAccountService.java
package com.bank.fd.service;

import com.bank.fd.dto.request.FdAccountCreateRequest;
import com.bank.fd.dto.response.FdAccountResponse;
import com.bank.fd.entity.FdStatement;
import com.bank.fd.entity.FdTransaction;

import java.util.List;

public interface FdAccountService {
    FdAccountResponse createAccount(FdAccountCreateRequest request, String createdBy);
    FdAccountResponse getAccount(String fdAccountNo);
    List<FdAccountResponse> getMyAccounts(String customerId);
    List<FdAccountResponse> getAllAccounts();
    List<FdTransaction> getTransactions(String fdAccountNo);
    List<FdStatement> getStatements(String fdAccountNo);
}
